clients/src/main/java/org/apache/kafka/clients/producer/internals/SenderMetricsRegistry.java

```diff
@@ -0,0 +1,124 @@
+    public MetricNameTemplate topicRecordSendRate;
+    public MetricNameTemplate topicByteRate;
+    public MetricNameTemplate topicCompressionRate;
+    public MetricNameTemplate topicRecordRetryRate;
+    public MetricNameTemplate topicRecordErrorRate;
+    public MetricNameTemplate batchSplitRate;
+
+    public SenderMetricsRegistry() {
+        this(new HashSet<String>());
+    }
+
+    public SenderMetricsRegistry(Set<String> tags) {
+
+        /* ***** Client level *****/
+        String metricGrpName = "producer-metrics";
+        
+        this.batchSizeAvg = new MetricNameTemplate("batch-size-avg", metricGrpName, "The average number of bytes sent per partition per-request.", tags);
+        this.batchSizeMax = new MetricNameTemplate("batch-size-max", metricGrpName, "The max number of bytes sent per partition per-request.", tags);
+        this.compressionRateAvg = new MetricNameTemplate("compression-rate-avg", metricGrpName, "The average compression rate of record batches.", tags);
+        this.recordQueueTimeAvg = new MetricNameTemplate("record-queue-time-avg", metricGrpName, "The average time in ms record batches spent in the record accumulator.", tags);
+        this.recordQueueTimeMax = new MetricNameTemplate("record-queue-time-max", metricGrpName, "The maximum time in ms record batches spent in the record accumulator.", tags);
+        this.requestLatencyAvg = new MetricNameTemplate("request-latency-avg", metricGrpName, "The average request latency in ms", tags);
+        this.requestLatencyMax = new MetricNameTemplate("request-latency-max", metricGrpName, "The maximum request latency in ms", tags);
+        this.recordSendRate = new MetricNameTemplate("record-send-rate", metricGrpName, "The average number of records sent per second.", tags);
+        this.recordsPerRequestAvg = new MetricNameTemplate("records-per-request-avg", metricGrpName, "The average number of records per request.", tags);
+        this.recordRetryRate = new MetricNameTemplate("record-retry-rate", metricGrpName, "The average per-second number of retried record sends", tags);
+        this.recordErrorRate = new MetricNameTemplate("record-error-rate", metricGrpName, "The average per-second number of record sends that resulted in errors", tags);
+        this.recordSizeMax = new MetricNameTemplate("record-size-max", metricGrpName, "The maximum record size", tags);
+        this.recordSizeAvg = new MetricNameTemplate("record-size-avg", metricGrpName, "The average record size", tags);
+        this.requestsInFlight = new MetricNameTemplate("requests-in-flight", metricGrpName, "The current number of in-flight requests awaiting a response.", tags);
+        this.metadataAge = new MetricNameTemplate("metadata-age", metricGrpName, "The age in seconds of the current producer metadata being used.", tags);
+        this.batchSplitRate = new MetricNameTemplate("batch-split-rate", metricGrpName, "The average number of batch splits per second", tags);
```

Comment:
I changed the description of this metric to be more readable. The previous one wasn't even in the docs, but in the code it said "The rate of record batch split"